apps/evm/src/containers/Layout/Header/MarketInfo/AddTokenToWalletDropdown/index.tsx

```diff
@@ -0,0 +1,82 @@
+import { Dropdown } from 'components/Dropdown';
+import useCopyToClipboard from 'hooks/useCopyToClipboard';
+import { useTranslation } from 'libs/translations';
+import { useAddTokenToWallet } from 'libs/wallet';
+import { useCallback } from 'react';
+import type { Token, VToken } from 'types';
+import { cn } from 'utilities';
+import type { ButtonProps } from '../../../../../components/Button';
+import { Icon } from '../../../../../components/Icon';
+import { TokenDropdownOption } from '../TokenDropdownOption';
+
+export interface AddTokenToWalletDropdownProps extends Omit<ButtonProps, 'onClick' | 'variant'> {
+  isUserConnected: boolean;
+  vToken: VToken;
+}
+
+const generateAddOrCopyTokenOption = (
+  token: VToken | Token,
+  label: string,
+  onClick: () => void,
+) => (
+  <span
+    className="flex items-center justify-start py-3 px-4 text-left text-sm font-semibold flex-row gap-2 grow min-w-[180px] cursor-pointer"
+    onClick={onClick}
+  >
+    <TokenDropdownOption token={token} label={label} />
+  </span>
```

Comment:
This whole JSX part can be the reusable component shared by the `AddTokenToWalletDropdown` and `GoToTokenContractDropdown` components, because although you currently use different classes for each they actually share the same UI so using either for both would give the same result.